### ISSUE TITLE
Update AppVeyor YAML template for resource modules

### ISSUE THREAD

**Issue by johlju** [PowerShell/DscResources#347]
I suggest we update the template for appveyor.yml at this location https://github.com/PowerShell/DscResources/blob/master/DscResource.Template/appveyor.yml with a more current one. 

I propose the below which uses the "default model"  and will run all the default common tests, with the option for the maintainer to opt-in for the rest of the test when so is possible.

```
#---------------------------------#
#      environment configuration  #
#---------------------------------#
version: 1.0.{build}.0
install:
    - git clone https://github.com/PowerShell/DscResource.Tests
    - ps: Write-Verbose -Message "PowerShell version $($PSVersionTable.PSVersion)" -Verbose
    - ps: Import-Module "$env:APPVEYOR_BUILD_FOLDER\DscResource.Tests\AppVeyor.psm1"
    - ps: Invoke-AppveyorInstallTask

#---------------------------------#
#      build configuration        #
#---------------------------------#

build: false

#---------------------------------#
#      test configuration         #
#---------------------------------#

test_script:
    - ps: |
        Invoke-AppveyorTestScriptTask -CodeCoverage -CodeCovIo -ExcludeTag @()
#---------------------------------#
#      deployment configuration   #
#---------------------------------#

# scripts to run before deployment
deploy_script:
    - ps: |
        Invoke-AppveyorAfterTestTask
```

**Comment by johlju** [comment]
This can be used by new resource modules, and old resource modules if they so choose. No existing resource modules should need to change to this.
Although I suggest this should be the minimum for new resource modules to be accepted into the DSC Resource Kit.

**Comment by johlju** [comment]
This was resolved in PR #348
